Castle/Internal/CASModel.m

```diff
@@ -56,4 +68,47 @@ - (NSString *)description
     return [NSString stringWithFormat:@"%@: %@", NSStringFromClass(self.class), [self JSONPayload]];
 }
 
+#pragma mark - NSSecureCoding
+
++ (BOOL)supportsSecureCoding
+{
+    return YES;
+}
+
+#pragma mark - Util
+
++ (BOOL)propertiesContainValidData:(NSDictionary *)dictionary
+{
+    // Check if dictionary is nil
+    if(!dictionary) {
+        return NO;
+    }
+
+    // Iterate through the contents and make sure there's no unsupported data types
+    for(id value in dictionary.allValues) {
```

Comment:
We won't support nested maps on the backend and all the values here will be cast to a String in Highwind.

1. We should remove support for nested Maps
2. Should we only allow strings or also allow Integer, Floats, Double & Nulls, but silently cast them to strings?

Reply:
1. Removed support for nested maps.
2. Integer, floats, double and nulls are silently cast to strings in highwind, so we can allow these types here. It might be better if we can stick with just strings because then we can change the type of the map to string values and not need to have it be object values.